Add `env` option to control setting the browserslist environment (#1012)
csstools/postcss-plugins: plugin-packs/postcss-preset-env/README.md

Any polyfills not explicitly enabled or disabled through `features` are
determined by the [`stage`](#stage) option.

### env

[PostCSS Preset Env] supports standard [browserslist] configuration, which
can be a `.browserslistrc` file, a `browserslist` key in `package.json`, or
`browserslist` environment variables.

> The `env` option is used by browserslist to determine the named environment
> it should use in the event that you have [multiple browserslist environments](https://github.com/browserslist/browserslist#configuring-for-different-environments) 
> configured. If not set, Browserslist will use the `production` environment.
> 
> ```js
> /* use the environment named `baseline`, instead of the default environment of  `production` */
> postcssPresetEnv({ env: 'baseline' })
> ```

Reviewer: ```suggestion
The `env` option is used by [browserslist] to determine the named environment
that should be used when you have [multiple browserslist environments](https://github.com/browserslist/browserslist#configuring-for-different-environments) 
configured. If not set, Browserslist will use the `production` environment.

```js
/* use the environment named `development`, instead of the default environment of `production` */
postcssPresetEnv({ env: 'development' })
```
```

`baseline` is a loaded term in this context now that Google and MDN are marketing something with this name.

Reviewer: _This suggestion was mangled by GitHub.
Better to apply it manually_

### browsers

> The browsers key shall override any existing [browserslist] configuration, and force the usage of certain browsers.

Reviewer: ```suggestion
The `browsers` option overrides any existing [browserslist] configuration.
```

### autoprefixer

> [PostCSS Preset Env] includes [autoprefixer]; the [`env`](#env) and [`browsers`](#browsers) optionwill be passed to it automatically.

Reviewer: ```suggestion
[PostCSS Preset Env] includes [autoprefixer]; the [`env`](#env) and [`browsers`](#browsers) options will be passed to it automatically.
```
